OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -615,16 +670,61 @@ public void SetPosition(Actor self, WPos pos)
 			self.World.UpdateMaps(self, this);
 
 			var altitude = self.World.Map.DistanceAboveTerrain(CenterPosition);
+
 			var isAirborne = altitude.Length >= Info.MinAirborneAltitude;
 			if (isAirborne && !airborne)
 				OnAirborneAltitudeReached();
 			else if (!isAirborne && airborne)
 				OnAirborneAltitudeLeft();
+
 			var isCruising = altitude == Info.CruiseAltitude;
 			if (isCruising && !cruising)
 				OnCruisingAltitudeReached();
 			else if (!isCruising && cruising)
 				OnCruisingAltitudeLeft();
+
+			FinishedMoving(self);
+		}
+
+		public void FinishedMoving(Actor self)
+		{
+			// Only make actor crush if it is on the ground
+			if (!self.IsAtGroundLevel())
+				return;
+
+			var actors = self.World.ActorMap.GetActorsAt(TopLeft).Where(a => a != self).ToList();
+			if (!AnyCrushables(actors))
+				return;
+
+			var notifiers = actors.SelectMany(a => a.TraitsImplementing<INotifyCrushed>().Select(t => new TraitPair<INotifyCrushed>(a, t)));
+			foreach (var notifyCrushed in notifiers)
+				notifyCrushed.Trait.OnCrush(notifyCrushed.Actor, self, Info.Crushes);
```

Comment:
We need to call `WarnCrush` first, probably when starting to land to match `Mobile.EnteringCell`, so that enemy units have a chance to run away.

Reply:
Added `EnteringCell` method to `Aircraft` and called upon landing initiation.

Reply:
Something is not working, they always die.

Reply:
Should be fixed now.